crates/accelerate/src/sparse_observable/lookup.rs

```diff
@@ -0,0 +1,294 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use num_complex::Complex64;
+// The lookup tables are pretty illegible if we have all the syntactic noise of `BitTerm`.
+use super::BitTerm::{self, *};
+
+/// Short-hand alias for [Complex64::new] that retains its ability to be used in `const` contexts.
+const fn c64(re: f64, im: f64) -> Complex64 {
+    Complex64::new(re, im)
+}
+
+/// The allowable items of `BitTerm`.  This is used by the lookup expansion; we need a const-safe
+/// way of iterating through all of the variants.
+static BIT_TERMS: [BitTerm; 9] = [X, Plus, Minus, Y, Right, Left, Z, Zero, One];
+
+/// Create a full binary lookup table for all valid combinations of `BitTerm`.
+macro_rules! expand_lookup_binary {
+    (static $table:ident: [$ty:ty], $generate:ident, $dummy:expr) => {
+        static $table: [$ty; 256] = {
```

Comment:
Just for my own sanity here: you use 256=2^8 here because you want to use the 4 bits of two bit terms to index -- but there are only 81 entries, right?

Reply:
Yeah, exactly. There are 81 explicit entries (though some are `None` themselves, so there's slightly fewer than 81 `Some` values), but the highest populated index is `0b10111011` (`(Left, Left)`, I think?), so most of the way to `0xff`.